kkm/kkm_main.c

```diff
@@ -779,13 +765,14 @@ static int kkm_get_native_cpuid(unsigned long arg)
 		ec[i].function = cpuid_functions[i];
 	}
 
-	if (copy_to_user(&(((struct kkm_cpuid *)arg)->entries[0]), ec,
-			 ec_entries_size)) {
+	if ((ret_val = kkm_to_user(&(((struct kkm_cpuid *)arg)->entries[0]), ec,
+				   ec_entries_size)) != 0) {
 		ret_val = -EFAULT;
 	}
 
 	kcpuid.reserved = 0;
-	if (copy_to_user((void *)arg, &kcpuid, sizeof(struct kkm_cpuid))) {
+	if ((ret_val = kkm_to_user((void *)arg, &kcpuid,
+				   sizeof(struct kkm_cpuid))) != 0) {
 		ret_val = -EFAULT;
 	}
```

Comment:
What is the point of calling kkm_to_user()?  If kkm_to_user() has a problem it returns -EFAULT which this code detects and sets ret_val to -EFAULT.  Seems like this is pointless.

Reply:
Should you also "goto error;" on copy failure?

Reply:
copy_to_user is a very commonly used  krnel function we cannot put breakpoint and handle the number of invocations.

kkm_to_user is a wrapper, a single point where we can put breakpoint. 

There is no point in assiging ret_val on function invocation(line 774). only a check is good enough.

Reply:
> Should you also "goto error;" on copy failure?

I will post a separate review for this. error from line  770 is being overwritten.
goto error is going to leak memory. Slightly different code needed.